sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -479,6 +479,14 @@ private static DoFnSignature parseSignature(Class<? extends DoFn<?, ?>> fnClass)
     return signature;
   }
 
+  private static Class<?> getDeclaringClass(Method onTimerMethod) {
```

Comment:
This is now needed? It seems very gross to pollute the main code with mockito stuff. Ideally we'd just fix the test to not use mockito. If this is just a recommended quick fix, that might be OK but file a Jira please to resolve it.

Reply:
Perform reflection on classes with mocks in question is very brittle and should not be done. This has been the compromise I have been using in other projects which is the least gross :cry: I don't have a JIRA account, do you mind filing an issue for me?

Reply:
Well, I'll turn that around and say that reflection is often OK when it is not fancy and mocking is generally not a good idea :-p. That said, both are worth avoiding if possible.

The relevant kind of reflection here is pretty elementary for annotation-driven API. I guess thinking about this case, the trigger of the issue is that we check that a couple classes are exactly the same class? Anathema to inheritance-based style but with the benefits of being simple and correct.

Created https://issues.apache.org/jira/browse/BEAM-8013

 If it is a different trigger of the problem, can correct the Jira.